### ISSUE TITLE
Support [] syntax

### ISSUE THREAD

**Issue by snyderp** [w3c/webidl2.js#126]
The IDL standard says that`[]` can be used to indicate a sequence of the proceeding token, ex

```
  [SameObject] readonly attribute DOMString[] types;
```

Is a sequence of `DOMString`s.

This is used in the wild in places like the HTML standard: ex https://www.w3.org/TR/html51/editing.html#the-datatransfer-interface

Currently the webidl2 parser does not understand this syntax:

The above line gives

```
Error: No name in attribute, line 1067 (tokens: '[] types;')
[
    {
        "type": "other",
        "value": "["
    },
    {
        "type": "other",
        "value": "]"
    },
    {
        "type": "whitespace",
        "value": " "
    },
    {
        "type": "identifier",
        "value": "types"
    },
    {
        "type": "other",
        "value": ";"
    }
]
```

If this is syntax you're interested in supporting, I'm happy to work on a PR for it.

**Comment by marcoscaceres** [comment]
This is invalid syntax as per latest WebIDL spec (the WebIDL spec on the TR is obsolete). We removed support for `[]` a while ago.

**Comment by snyderp** [comment]
Ah, I see!  Is there any interesting in enabling such legacy syntax in the parser behind a "legacy" flag or similar, so that the parser can be used to parse older standards?

**Comment by marcoscaceres** [comment]
No, unfortunately, as we want to make sure people update their specs. We have tried to put systems in place that give spec editors plenty of time to update their specs (e.g., ~1 year), and generally don't yank features until all (W3C/WHATWG) specs are been updated. 

Is your spec in a private repo?

**Comment by snyderp** [comment]
Ah, I understand, thanks!

I'm only pulling public W3C (and related) specs, to do things like tracking feature growth / change in browsers.  So, I keep running into situations where I need to read old standards (which is where the torrent of issues are coming from, ex #128).  I'll only open up issues for the current WebIDL spec going forward.

In the meantime, I'll pull the relevant WebIDL documents into my own repo and edit them to bring them into compliance with the parser.  Thanks!

**Comment by snyderp** [comment]
Sounds good, will do.  Already found one issue in the WebUSB spec 

https://github.com/WICG/webusb/issues/118
